src/ParameterDefinition.php

```diff
@@ -187,19 +187,33 @@ private function resolveUnionType(ContainerInterface $container)
                     /** @psalm-suppress PossiblyNullReference */
                     $typeName = $this->parameter->getDeclaringClass()->getName();
                 }
+
```

Comment:
```suggestion
$error = null;

try {
    /** @var mixed */
    $result = $container->get($typeName);
} catch (Throwable $t) {
    $error = $t;
}

if ($error === null) {
    /** @var mixed $result Exist, because $resolved is true */
    if (!$result instanceof $typeName) {
        $actualType = $this->getValueType($result);
        throw new InvalidConfigException(
            "Container returned incorrect type \"$actualType\" for service \"$class\"."
        );
    }
    return $result;
}
```

Reply:
It's incorrect. `$error` can be already create for previous type.

Reply:
You can add the null error inside the try and fixed.

Reply:
`$error` used outside of foreach.
`$result` can be null thet return from container (`$container->get()` return mixed).